### PR TITLE
Update component documentation

### LINKED ISSUE #2772: Make the component documentation clear and understandable
### Goal:
**The style guide is _clear and understandable_**
_Other issues connected to this goal: #2773_
____

### User stories
- As a UX designer I want to feel confident that I understand what a component is used for, and am choosing to use a component consistently with the project’s design vernacular. 

**Steps for this issue's completion:**
- [x] Create a content model or template for component documentation
- [x] Identify specific gaps and areas for expansion in component documentation.
- [ ] Make the changes!

### PR DESCRIPTION
**Pull request by adborden**
Fixes #2772

[:sunglasses: PREVIEW](https://federalist-proxy.app.cloud.gov/preview/onrr/doi-extractives-data/styleguide-docs/styleguide/)

Changes proposed in this pull request:

- Update documentation for each component.
- Split components into individual files.
- Document best practices for component documentation.

### CODE REVIEW

**Comment by adborden** [comment]
I had a quick chat with @mcharg and I think this is ready for a review. It doesn't update all the components yet because I want to get more feedback from the team.

- Is this the right level of documentation?
- Is this information useful?
- Is this information reasonable to produce (not too burdensome)?

**Comment by adborden** [comment]
Rebased to resolve merge conflicts.

File: nrrd-design-system/components/components/accordions/accordions--mobile-menu.hbs [hunk 1]
```diff
@@ -0,0 +1,24 @@
+<main id="foo" class="container-page-wrapper layout-state-pages">
+  <div id="top" aria-label="anchor for mobile navigation" class="mobile-nav-anchor"></div>
+  <div is="mobile-nav" class="mobile-nav sticky stuck">
+    <button is="aria-toggle" aria-controls="mobile-nav-content" aria-expanded="false" type="button">
+      <div class="flex-row">
+        <h3 class="mobile-nav-header flex-row-flex">{{ label }}</h3>
+        <span class="hide-expanded flex-row-icon">
+          <icon class="icon icon-chevron-sm-down"></icon>
```
Comment (brentryanjohnson) [inline nrrd-design-system/components/components/accordions/accordions--mobile-menu.hbs]: This chevron appears much larger relative to its text in the style guide than it does on the site. Any ideas why @adborden?

## Style guide
style guide accordion menu

## Production site
site accordion menu

**Comment by adborden** [comment]
@master12 You probably want to get this merged before really digging into the [horizontal bar graphs](https://github.com/ONRR/doi-extractives-data/issues/3027). It has both a [component README template](https://github.com/ONRR/doi-extractives-data/pull/3035/files#diff-ddd20bed32e9c9effcb43785418bfd9c) and the [how-to](https://github.com/ONRR/doi-extractives-data/pull/3035/files#diff-e1723ebf3dd1f6389735d9f3c05acd18) which has some fractal tips.

**Pull request merged**